[CI] openblas build failed in static build

I notice the openblas 0.3.5 build failed in static build but the CI still passed.

```
../libopenblasp-r0.3.5.a(cblas_sgemm.o): In function `cblas_sgemm':
gemm.c:(.text+0x57f): undefined reference to `sgemm_kernel_direct_performant'
gemm.c:(.text+0x5d8): undefined reference to `sgemm_kernel_direct'
collect2: error: ld returned 1 exit status
make[1]: *** [xscblat3] Error 1
make[1]: *** Waiting for unfinished jobs....
make: *** [tests] Error 2
++++ make --quiet -j 72 PREFIX=/work/mxnet/staticdeps install
make[1]: warning: -jN forced in submake: disabling jobserver mode.
Generating openblas_config.h in /work/mxnet/staticdeps/include
Generating f77blas.h in /work/mxnet/staticdeps/include
Generating cblas.h in /work/mxnet/staticdeps/include
Copying LAPACKE header files to /work/mxnet/staticdeps/include
Copying the static library to /work/mxnet/staticdeps/lib
Copying the shared library to /work/mxnet/staticdeps/lib
install: cannot stat 'libopenblasp-r0.3.5.so': No such file or directory
make[1]: *** [install] Error 1
make: *** [install] Error 2
```
For example:
http://jenkins.mxnet-ci.amazon-ml.com/blue/organizations/jenkins/mxnet-validation%2Funix-cpu/detail/PR-15983/3/pipeline/297

--- a/tools/dependencies/openblas.sh
+++ b/tools/dependencies/openblas.sh
@@ -18,9 +18,8 @@
 # under the License.
 
 # This script builds the static library of openblas that can be used as dependency of mxnet.
-set +e # This script throws an error but otherwise works
-set -x
-OPENBLAS_VERSION=0.3.5
+set -ex
+OPENBLAS_VERSION=0.3.7
 if [[ ! -e $DEPS_PATH/lib/libopenblas.a ]]; then
     # download and build openblas
     >&2 echo "Building openblas..."
@@ -32,8 +31,13 @@ if [[ ! -e $DEPS_PATH/lib/libopenblas.a ]]; then
     pushd .
     cd $DEPS_PATH/OpenBLAS-$OPENBLAS_VERSION
 
-    $MAKE DYNAMIC_ARCH=1 NO_SHARED=1 USE_OPENMP=1
+    # Adding NO_DYNAMIC=1 flag causes make install to fail
+    $MAKE DYNAMIC_ARCH=1 USE_OPENMP=1
     $MAKE PREFIX=$DEPS_PATH install
+
+    # Manually removing .so to avoid linking against it
+    rm $DEPS_PATH/lib/libopenblasp-r${OPENBLAS_VERSION}.so
+
     popd
     ln -s libopenblas.a $DEPS_PATH/lib/libcblas.a
     ln -s libopenblas.a $DEPS_PATH/lib/liblapack.a

